bootstrap/secret/secure.go

```diff
@@ -190,3 +196,69 @@ func (p *SecureProvider) DefaultTokenExpiredCallback(expiredToken string) (repla
 
 	return reReadToken, true
 }
+
+func (p *SecureProvider) LoadServiceSecrets(filePath string) error {
+	contents, err := os.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("seeding secrets failed: %s", err.Error())
+	}
+
+	data, seedingErrs := p.seedSecrets(contents)
+
+	if err := os.WriteFile(filePath, data, 0); err != nil {
+		return fmt.Errorf("seeding secrets failed: unable to overwrite file with secret data removed: %s", err.Error())
```

Comment:
Has seeding secrets really failed in this case? It seems to me its possible for p.seedSecrets to succeed and actually provision the supplied secrets, yet if the file isn't writable for whatever reason, we tell the calling service that the operation failed.

Maybe you should just log a warning in the case of the write failure?

Reply:
HUmm, i get where you are coming from, but I think @bnevis-i would prefer this to be an error since it didn't rewrite the file w/o the secretData. No one will see the warning message since the secrets get seeded.

Reply:
Should we also remove the secrets then too (if possible) so that we're handling this as a single transaction?

Reply:
> Should we also remove the secrets then too (if possible) so that we're handling this as a single transaction?

No, it doesn't hurt to store them again once the write issue is resolved.

Reply:
@bnevis-i , please comment on error and fail vs. logging warning when writing file back out w/o the secret data.

Reply:
Let's:
 - If secrets are there, import them
 - Attempt to scrub them unless told not to
 - Fail if either can't import or can't scrub when we're told to.

Reply:
@tonyespy , @bnevis-i , I am adding the `DisableScrubSecretsFile` setting to SecretStoreInfo to allow users to control this. Default will be false so secrets get scrubbed unless they explicitly change the setting.